File: dynamic_listings/javascript/default/computer_science/smart_array_init.js
// ...
const tyler_array_len = 64;
let tyler = [];
for (let i = 0; i < tyler_array_len; i++) {
    tyler.push(CharacterExpression.neutral);
}
// Here we take care of the exceptions
tyler[33] = CharacterExpression.angry;
tyler[34] = CharacterExpression.sad;
// ... Next character ...

File: dynamic_listings/rust/default/computer_science/smart_array_init.rs
# #![allow(dead_code, unused)]
#
# enum CharacterExpression {
#     Neutral,
#     Angry,
#     Sad,
# }
# fn main() {
// ...
let tyler_array_len = 64;
let mut tyler = Vec::new();
for _ in 0..64 {
    tyler.push(CharacterExpression::Neutral);
}
// Here we take care of the exceptions
tyler[33] = CharacterExpression::Angry;
tyler[34] = CharacterExpression::Sad;
// ... Next character ...
# }
